sensor/kubernetes/listener/resources/endpoints.go

```diff
@@ -74,7 +74,7 @@ func (m *endpointManagerImpl) addEndpointDataForPod(pod *v1.Pod, data *clusteren
 	podIP := net.ParseIP(pod.Status.PodIP)
 	// Do not register the pod if it is using the host network (i.e., pod IP = node IP), as this causes issues with
 	// kube-proxy connections.
-	if !pod.Spec.HostNetwork && podIP.IsValid() {
+	if allowHostNetworkPodIPsInEntitiesStore.BooleanSetting() || (!pod.Spec.HostNetwork && podIP.IsValid()) {
```

Comment:
The left part of the `or` does not check if `podIP.IsValid()`

Reply:
> The left part of the or does not check if podIP.IsValid()

```suggestion
	if (allowHostNetworkPodIPsInEntitiesStore.BooleanSetting() || !pod.Spec.HostNetwork) && podIP.IsValid() {

```

Reply:
That is no a trivial to catch - nice! Will fix 👍